test/unit/auth/user-import-builder.spec.ts

```diff
@@ -555,6 +601,77 @@ describe('UserImportBuilder', () => {
           new UserImportBuilder(noHashUsers, validOptions as any, userRequestValidator);
       expect(userImportBuilder.buildRequest()).to.deep.equal(expectedRequest);
     });
+
+    it('should return expected request with no multi-factor fields when not available', () => {
+      const noMultiFactorUsers: UserImportRecord[] = [
+        {uid: '1234', email: 'user@example.com', multiFactor: null},
+        {uid: '5678', phoneNumber: '+16505550101', multiFactor: {enrolledFactors: []}},
+      ];
+      const expectedRequest = {
+        users: [
+          {localId: '1234', email: 'user@example.com'},
+          {localId: '5678', phoneNumber: '+16505550101'},
+        ],
+      };
+      const userImportBuilder =
+          new UserImportBuilder(noMultiFactorUsers, validOptions as any, userRequestValidator);
+      expect(userImportBuilder.buildRequest()).to.deep.equal(expectedRequest);
+    });
+
+    it('should ignore users with invalid second factor enrollment time', () => {
+      const invalidMultiFactorUsers: UserImportRecord[] = [
+        {
+          uid: '1234',
+          multiFactor: {
+            enrolledFactors: [
+              {
+                uid: 'enrolledSecondFactor1',
+                phoneNumber: '+16505557348',
+                displayName: 'Spouse\'s phone number',
+                factorId: 'phone',
+                enrollmentTime: 'invalid',
+              },
+            ],
+          },
+        },
+        {uid: '5678', phoneNumber: '+16505550102'},
+      ];
+      const expectedRequest: UploadAccountRequest = {
+        users: [
+          {localId: '5678', phoneNumber: '+16505550102'},
+        ],
+      };
+      const userImportBuilder =
+          new UserImportBuilder(invalidMultiFactorUsers, validOptions as any, userRequestValidator);
+      expect(userImportBuilder.buildRequest()).to.deep.equal(expectedRequest);
+    });
+
+    it('should ignore users with unsupported second factors', () => {
+      const invalidMultiFactorUsers: any = [
+        {
+          uid: '1234',
+          multiFactor: {
+            enrolledFactors: [
+              {
+                uid: 'enrolledSecondFactor1',
+                secret: 'SECRET',
+                displayName: 'Google Authenticator on personal phone',
+                factorId: 'totp',
```

Comment:
Expected to throw. 

It looks like there's some pre-existing logic in the UserImportBuilder to handle these errors, and ignore them. May be we should at least log a warning when this happens, so a simple typo in the user code doesn't go completely unnoticed.

Reply:
Same here. The client side errors are cached and will be injected in the response alongside server side response. Check test `should return the expected response for import with client side errors`. I added a test for this too.